### DISCUSSION TITLE
How to connect prisma model with entry created by Hasura ?

### DISCUSSION THREAD

**Discussion by rootandleaves** [prisma/prisma#3911] [Q&A] [answered]
I am using Hasura for postgis. so basically i have following model which needs for postgis data.

#Location
`
```prisma
model Location {
  id              String    @id @default(cuid())
  createdAt       DateTime  @default(now())
  updatedAt       DateTime  @default(now()) @updatedAt
  Account__ufk__  String    @unique @map("__Account__ufk__")
  Locations__fk__ String    @map("__Locations__fk__")
  Account         Account   @relation(fields: [Account__ufk__], references: [id])
  Locations       Locations @relation(fields: [Locations__fk__], references: [id])
}

model Locations {
  id          String     @id @default(cuid())
  createdAt   DateTime   @default(now())
  updatedAt   DateTime   @default(now()) @updatedAt
  // This type is currently not supported.
  point    point
  // This type is currently not supported.
  previous point?
  Location    Location[]
}
```
`

Now i make insertion as follows. But prisma fails to connect data which is already inserted in database.
if i do prisma.findOne with _id than prisma also do return data. but always fails to connect relationship.
`
```ts
const _id = CUID()
await Fetch(
          http://localhost:8080/v1alpha1/graphql,
          {
            method: 'POST',
            body: JSON.stringify({
              'query': `
               mutation insert_Locations($id: String, $point: point) {
                  insert_Locations(objects:{id: $id, point: $point }) {
                    returning {
                      id
                      createdAt
                      updatedAt
                    }
                  }
              }
            `,
              'variables': { 'id': _id, 'point': `${latitude}, ${longitude}` },
              'operationName': 'insert_Locations'
            })
          }
        )

await Context.DataBase.location.create({
            'data': {
              'Locations': {
                'connect': {
                  'id': _id <---- Always fails with error described below.
                }
              },
              'Account': {
                'connect': {
                  'id': Context.Session.user.id
                }
              }
            }
          })
```
`

Error:
`
{"details":"Error for binding '2': AssertionError(\"[Query Graph] Expected a valid parent ID to be present for a nested connect on a one-to-one relation.\")"}
`

**Comment by ryands17** [comment, marked as answer]
It seems that the location is not inserted immediately after you apply your mutation. Could you check if Prisma can fetch the location first instead of connecting it as you're doing above?

**Reply by rootandleaves** [reply]
i did it...and Prisma.findOne returns object...thats why i am confused.

**Reply by ryands17** [reply]
Would it be possible for you to provide a sample reproduction repo so that I can check from my end?

**Reply by pantharshit00** [reply]
@rootandleaves Also try logging the SQL generated by Prisma and the variables it is getting.

Here how you can do it:
```ts
// enable logging events in the client constructor
const prisma = new PrismaClient({
  log: [
    {
      emit: "event",
      level: "query",
    },
  ],
});

prisma.$on("query", async (e) => {
  // write queries and variables to a file, you can also choose to log them
  await fs.promises.appendFile(
    __dirname + "/test.sql",
    `${e.query} ${e.params} \n`
  );
});
```

**Reply by rootandleaves** [reply]
ok great

**Reply by ryands17** [reply]
Hey @rootandleaves did you find the issue?

**Reply by rootandleaves** [reply]
No @ryands17 :(

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
